### ISSUE TITLE
DPRReaderTokenizers returns, for multiple passages given, only the tokens & masks of one passage 

### ISSUE THREAD

**Issue by omarsou** [huggingface/transformers#8705]
## Environment info
     
- `transformers` version: 3.5.1
- Platform: Colab Notebook
- Python version: 3.6.9
- PyTorch version (GPU?): 1.7.0
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help
 tokenizers: @mfuntowicz 
 -->

## Information

Model I am using : DPRReaderTokenizer

The problem arises when using:
DPRReaderTokenizer, instead of returning as many tensor as passages, he returns only one (On the documentation it must return (n_passages, sequence_length), but it returns (1, sequence_length) on basic examples.

The tasks I am working on is:
* Tokenization with DPRReaderTokenizer on multiple passages (texts)

## To reproduce

from transformers import AlbertTokenizer, AlbertForQuestionAnswering, DPRReader, DPRReaderTokenizer, AutoTokenizer
import torch
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

tokenizer_DPR = DPRReaderTokenizer.from_pretrained('facebook/dpr-reader-single-nq-base')
model_DPR = DPRReader.from_pretrained('facebook/dpr-reader-single-nq-base', return_dict=True).cuda()

encoded_inputs = tokenizer_DPR(
         questions=["What is Transformers?"],
         titles=['Attention is all you need', 'One famous library'],
         texts=['Attention is a new mechanism designed to improve the performance of the seq2seq models', 'One of the most
famous NLP library is called Transformers' ],
         padding=True,
         return_tensors='pt'
)

encoded_inputs
{'input_ids': tensor([[  101,  2054,  2003, 19081,  1029,   102,  3086,  2003,  2035,  2017,
          2342,   102,  3086,  2003,  1037,  2047,  7337,  2881,  2000,  5335,
          1996,  2836,  1997,  1996,  7367,  4160,  2475,  3366,  4160,  4275]]), 'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
         1, 1, 1, 1, 1, 1]])}

## Remarks
The expected outputs is two tensors ... and I got only one .. 👎

**Comment by omarsou** [comment]
Hello, 
A little update, I think I fixed this issue. 
The tokenizer returns a tensor of shape (n_passages, n_sequence_length) but only because I have duplicated the question like [questions] * n_passages. It was not clear on the documentation since I thought it was automatically done.
